server/plugin.go

```diff
@@ -146,6 +156,32 @@ func (p *Plugin) OnPluginClusterEvent(c *plugin.Context, ev model.PluginClusterE
 	p.HandleClusterEvent(ev)
 }
 
+func (p *Plugin) MessageWillBePosted(c *plugin.Context, post *model.Post) (*model.Post, string) {
+	// If not enabled in config, ignore.
+	if !p.getConfiguration().EnableCodePreview {
+		return nil, ""
+	}
+
+	if post.UserId == "" {
+		return nil, ""
+	}
+
+	msg := post.Message
+	info, err := p.getGitlabUserInfoByMattermostID(post.UserId)
+	if err != nil {
+		p.API.LogError("error in getting user info", "error", err.Message)
+		return nil, ""
+	}
```

Comment:
What if the user simply isn't connected to GitLab? We don't want to log this as an "error" in this case. Also, we'll want to be more specific with these error messages, saying that we are processing a post's permalinks, to differentiate from an intentional action related to GitLab.

I'm thinking all `LogError`'s in this PR should be changed to `LogDebug` due to the passive nature of this feature

Reply:
Done